manager/volume.go

```diff
@@ -742,6 +746,120 @@ func (m *VolumeManager) EngineUpgrade(volumeName, image string) (v *longhorn.Vol
 	return v, nil
 }
 
+func (m *VolumeManager) checkVolumeNotInMigration(volumeName string) error {
+	v, err := m.Get(volumeName)
+	if err != nil {
+		return err
+	}
+	if v.Spec.MigrationNodeID != "" {
+		return fmt.Errorf("cannot operate during migration")
+	}
+	return nil
+}
+
+func (m *VolumeManager) MigrationStart(name, nodeID string) (v *longhorn.Volume, err error) {
```

Comment:
Refactored the manager/volume.go and removed the dedicated migrationStart/Confirm/Rollback functions since they are no longer needed now that we use the volume attach&detach for handling of this.